WSLBlobNFS/wsl2_linux_script.sh

```diff
@@ -60,54 +166,184 @@ function onetime_samba_setup ()
 
 # Mount NFS
 # $1: mount command
+# $2: mount path
 # To-do: Return the result of the mount command
 function mount_nfs ()
 {
-    echo "Got $# number of args: $*"
-    # execute the mount command passed as the first argument
-    $1
+    if [[ $# != 2 ]]; then
+        eecho "Usage: mount_nfs <mount command> <mount path>"
+        exit 1
+    fi
+
+    mountcommand=$1
+    mountpath=$2
+
+    # Execute the mount command
+    eval $mountcommand
+
+    if [[ $? != 0 ]]; then
+        eecho "Failed to mount NFS share with: $mountcommand."
+        exit 1
+    fi
 
     # Set the read ahead to 16MB
-    echo "Setting read ahead to 16MB"
-    cd $2
-    echo 16384 > /sys/class/bdi/0:$(stat -c "%d" .)/read_ahead_kb
-    cd -
+    vecho "Setting read ahead to 16MB"
+    echo 16384 > /sys/class/bdi/0:$(stat -c "%d" $mountpath)/read_ahead_kb
 
-    echo "Mounted NFS share."
+    secho "Mounted NFS share using $mountcommand"
+}
+
+# Mount SMB and NFS share
+# $1: mount parameter type
+# $2: mount parameter
+# $3: temp file path
+function mount_share ()
+{
+    # Create the mountpath, sharename, and mountCommand
+    mountPath=""
+    shareName=""
+    mountCommand=""
+
+    if [[ $mountparametertype == "command" ]]; then
+        # MountCommand: mount -t nfs -o vers=3,proto=tcp <account-name>.blob.core.windows.net:/<account-name>/<container-name> /mnt/<path>
+        mountCommand=$mountparameter
+        vecho "Mount command is: $mountCommand"
+
+        # Split the mount command into tokens
+        IFS=' ' read -ra nametokens <<< "$mountCommand"
+
+        # last element is the mount point
+        mountPath=${nametokens[-1]}
+
+        # check if the last element is a path
+        if [[ ${mountPath:0:1} == "/" ]]; then
+            # check if the path exists
+            if [[ ! -d $mountPath ]]; then
+                mkdir -p $mountPath
+                vecho "Created $mountPath"
+            else
+                vecho "Mount point $mountPath already exists"
+            fi
+        else
+            eecho "Mount point $mountPath is not a path"
+            exit 1
+        fi
+
+        mpath=${mountPath:1}
+
+        # replace all / with -
+        shareName=${mpath//\//-}
+    else
+        # RemoteHost: <account-name>.blob.core.windows.net:/<account-name>/<container-name>
+
+        # A random non existent mount path to mount the NFS share
+        randomnumber=$RANDOM
+        mountPath="/mnt/nfsv3share-$randomnumber"
+
+        while [[ -e $mountPath ]]; do
+            randomnumber=$RANDOM
+            mountPath="/mnt/nfsv3share-$randomnumber"
+        done
+
+        mkdir -p $mountPath
+        vecho "Created $mountPath"
+
+        mountCommand="mount -t nfs -o vers=3,proto=tcp $mountparameter $mountPath"
+        shareName="nfsv3share-$randomnumber"
+    fi
+
+    # Check if the mount path is already mounted
+    mount | grep -q $mountPath
```

Comment:
to prevent this from wrongly matching a different mountPath with the same prefix f.e. /mnt/poin matches /mnt/point, let's do a more specific search "on $mountPath type" 
or better to use the mountpoint command
mountpoint $mountPath